Bottom: f97f83c545f70dd6a011ad916a98bf48a5ffa59c
Top:    bdd751328653f9acc9353d8f80f48d12b0919e04
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-05-09 11:56:21 -0700

wifi: Don't spam logs with 'Found new beacon' messages.

We saw logs fill with this (at very high speeds):

cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0
cfg80211: Found new beacon on frequency: 5745 MHz (Ch 149) on wiphy0

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/wireless/reg.c b/net/wireless/reg.c
index cdf96ae..b82fbf7 100644
--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -2672,10 +2672,11 @@ int regulatory_hint_found_beacon(struct wiphy *wiphy,
 	if (!reg_beacon)
 		return -ENOMEM;
 
-	pr_debug("Found new beacon on frequency: %d MHz (Ch %d) on %s\n",
-		 beacon_chan->center_freq,
-		 ieee80211_frequency_to_channel(beacon_chan->center_freq),
-		 wiphy_name(wiphy));
+	if (printk_ratelimit())
+		pr_debug("Found new beacon on frequency: %d MHz (Ch %d) on %s\n",
+			 beacon_chan->center_freq,
+			 ieee80211_frequency_to_channel(beacon_chan->center_freq),
+			 wiphy_name(wiphy));
 
 	memcpy(&reg_beacon->chan, beacon_chan,
 	       sizeof(struct ieee80211_channel));
